Frontend/src/Components/BatteryCells/BatteryPack.js

```diff
@@ -0,0 +1,121 @@
+                dataMax={CONSTANTS.pack_voltage.MAX * CONSTANTS.pack_current.MAX}
+                bg={headerBg}
+                DecimalPoint='3'
+            />
+            <DataPack
+                dataTitle="Pack Temperature"
+                dataValue={props.data?.pack_temp[0] ?? -1.0}
+                dataUnit="&#8451;"
+                dataMin={CONSTANTS.pack_temp.MIN}
+                dataMax={CONSTANTS.pack_temp.MAX}
+                DecimalPoint='3'
+            />
+            <DataPack
+                dataTitle="Fan Speed"
+                dataValue={props.data?.fan_speed[0] ?? -1.0}
+                dataUnit={CONSTANTS.fan_speed.UNIT}
+                dataMin={((props.data?.pack_temp[0] ?? -1.0) < (20 + 5 * CONSTANTS.fan_speed.MAX)) ?
+                         Math.floor(((((props.data?.pack_temp[0] ?? -1.0) >= 20) ? props.data?.pack_temp[0] : 15) - 15) / 5) :
+                         CONSTANTS.fan_speed.MAX}
+                dataMax={CONSTANTS.fan_speed.MAX}
+                bg={headerBg}
+                DecimalPoint='0'
+            />
+            <DataPack
+                dataTitle="Pack Resistance"
+                dataValue={props.data?.pack_resistance[0] ?? -1.0}
+                dataUnit="m&#937;"
+                dataMin={CONSTANTS.pack_resistance.MIN}
+                dataMax={CONSTANTS.pack_resistance.MAX}
+                DecimalPoint='3'
+            />
+            <DataPack
+                dataTitle="Avg. Cell Int. Resist"
+                dataValue={props.data?.avg_cell_resistance[0] ?? -1.0}
+                dataUnit="m&#937;"
+                dataMin={CONSTANTS.avg_cell_resistance.MIN}
+                dataMax={CONSTANTS.avg_cell_resistance.MAX}
+                bg={headerBg}
+                DecimalPoint='3'
+            />
+            <DataPack
+                dataTitle="Pack Capacity"
+                dataValue={props.data?.pack_capacity[0] ?? -1.0}
+                dataUnit={CONSTANTS.pack_capacity.UNIT}
+                dataMin={CONSTANTS.pack_capacity.MIN}
+                dataMax={CONSTANTS.pack_capacity.MAX}
+                DecimalPoint='3'
+            />
+            <DataPack
+                dataTitle="Adap. Total Capacity"
+                dataValue={props.data?.adaptive_total_capacity[0] ?? -1.0}
+                dataUnit={CONSTANTS.adaptive_total_capacity.UNIT}
+                dataMin={CONSTANTS.adaptive_total_capacity.MIN}
+                dataMax={CONSTANTS.adaptive_total_capacity.MAX}
+                bg={headerBg}
+                DecimalPoint='3'
+            />
+            <DataPack
+                dataTitle="Pack State of Health"
+                dataValue={props.data?.soh[0] ?? -1.0}
+                dataUnit={CONSTANTS.soh.UNIT}
+                dataMin={CONSTANTS.soh.MIN}
+                dataMax={CONSTANTS.soh.MAX}
+                DecimalPoint='3'
+                borderBottomWidth={0}
+            />

```

Comment:
You could remove multiple, but it would probably look weird after removing a few. You can add up to 2 before it breaks the layout, so no, it doesn't really dynamically render. I'm not sure about reading from the format or another file/JSON. It would be doable, but I'm not sure that it would be worth the effort. In any case, it would definitely warrant a separate task.

Reply:
cool we'll table that for later